src/Controller/SantaController.php

```diff
@@ -137,15 +125,27 @@ public function message(Rudolph $rudolph, FormFactoryInterface $formFactory, Req
             return new RedirectResponse($this->router->generate($application->getAuthenticationRoute()));
         }
 
+        $errors = [];
+
         $session = $request->getSession();
-        $availableUsers = $session->get('available-users', []);
-        $selectedUsers = $session->get('selected-users', []);
-        $message = $session->get('message');
-        $notes = $session->get('notes');
-        $options = $session->get('options', []);
-
-        $builder = $formFactory->createBuilder(MessageType::class, ['message' => $message], [
-            'selected-users' => $selectedUsers,
+
+        /** @var Config $config * */
+        $config = $session->get('config');
+
+        $selectedUsersAsArray = $config->getSelectedUsers();
+
+        $notes = array_filter($config->getNotes(), function ($userIdentifier) use ($selectedUsersAsArray) {
```

Comment:
Could you add a comment to explain that we clean/prepare the notes according to the selected users here

Reply:
Added ^^